tests/invalid_models_tests/test_relative_fields.py

```diff
@@ -120,12 +120,6 @@ class ThroughModel(models.Model):
                 obj=field,
                 id='fields.W341',
             ),
-            DjangoWarning(
```

Comment:
Do we need to remove `ThroughModel` and `through='ThroughModel'` in `ModelM2M.m2m`?

Reply:
Yes, I'd remove them from this test method and add a separate test, e.g.
```python
    def test_many_to_many_with_limit_choices_through_no_warning(self):
        class Model(models.Model):
            name = models.CharField(max_length=20)

        class ModelM2M(models.Model):
            m2m = models.ManyToManyField(
                Model,
                limit_choices_to={'name': 'test_name'},
                through='ThroughModel',
                through_fields=('modelm2m', 'model'),
            )

        class ThroughModel(models.Model):
            model = models.ForeignKey('Model', models.CASCADE)
            modelm2m = models.ForeignKey('ModelM2M', models.CASCADE)

        self.assertEqual(ModelM2M.check(), [])
```